google/cloud/documentai_toolbox/wrappers/page.py

```diff
@@ -411,9 +603,73 @@ def __post_init__(self):
+            text=self.text,
+            paragraphs=self.paragraphs,
         )
-        self.blocks = _get_blocks(blocks=self.documentai_page.blocks, text=self.text)
         self.tables = tables
+
+    def to_hocr(self, filename):
+        r"""Exports a string hOCR version of the documentai.Document.Page.
+
+        Args:
+            filename (str):
+                Required. The filename of the original file.
+
+        Returns:
+            str:
+                A string hOCR version of the documentai.Document.Page.
+        """
+        f = ""
+        dimensions = self.documentai_page.dimension
+        pidx = self.documentai_page.page_number
+        page_bounding_box = _get_hocr_bounding_box(
+            element_with_layout=(self.documentai_page), dimensions=(dimensions)
+        )
+        f += f"<div class='ocr_page' lang='unknown' title='image \"{filename}\";{page_bounding_box}'>\n"
+        for bidx, block in enumerate(self.blocks):
+            block_bounding_box = _get_hocr_bounding_box(
+                element_with_layout=(block.documentai_block), dimensions=(dimensions)
+            )
+            f += f"<span class='ocr_carea' id='block_{pidx}_{bidx}' title='{block_bounding_box}'>\n"
+            for paridx, paragraph in enumerate(block.paragraphs):
+                paragraph_bounding_box = _get_hocr_bounding_box(
+                    element_with_layout=(paragraph.documentai_paragraph),
+                    dimensions=(dimensions),
+                )
+                f += f"<span class='ocr_par' id='par_{pidx}_{bidx}_{paridx}' title='{paragraph_bounding_box}'>\n"
+                for lidx, line in enumerate(paragraph.lines):
+                    line_bounding_box = _get_hocr_bounding_box(
+                        element_with_layout=(line.documentai_line),
+                        dimensions=(dimensions),
+                    )
+                    line_text = (
+                        line.text.replace("&", "&amp;")
+                        .replace("<", "&lt;")
+                        .replace(">", "&gt;")
+                    )
+                    f += f"<span class='ocr_line' id='line_{pidx}_{bidx}_{paridx}_{lidx}' title='{line_bounding_box}'>{line_text}</span>\n"
+                    for tidx, token in enumerate(line.tokens):
+                        token_bounding_box = _get_hocr_bounding_box(
+                            element_with_layout=(token.documentai_token),
+                            dimensions=(dimensions),
+                        )
+                        word_text = (
+                            token.text.replace("&", "&amp;")
+                            .replace("<", "&lt;")
+                            .replace(">", "&gt;")
+                        )
+                        f += f"<span class='ocrx_word' id='word_{pidx}_{bidx}_{paridx}_{lidx}_{tidx}' title='{token_bounding_box}'>{word_text}</span>\n"
+                f += "</span>\n"
+            f += "</span>\n"
+        f += "</div>\n"
```

Comment:
I wonder if we can omit these newlines, and at the end optionally call an xml-formatter (if the user wants the xml string to look pretty).

Reply:
yea the newlines can be omitted the looking pretty doesn't matter to the initial customer